### ISSUE TITLE
FTBFS on 5.38.1/5.36.2/5.34.2

### ISSUE THREAD

**Issue by zakame** [Perl/perl5#21671]
Hi, thanks for the recent Perl releases! 🎉

However I think it seems a bit scuffed - while preparing for the https://github.com/Perl/docker-perl updates I'm seeing this test failure across all three releases:

```
% docker buildx build -t perl:5.38.1 .
[...]
146.0 
146.0 Test Summary Report
146.0 -------------------
146.0 porting/regen.t                                                    (Wstat: 65280 (exited 255) Tests: 44 Failed: 1)
146.0   Failed test:  10
146.0   Non-zero exit status: 255
146.0 Files=2766, Tests=1183930, 84 wallclock secs (55.57 usr  8.56 sys + 368.19 cusr 36.69 csys = 469.01 CPU)
146.0 Result: FAIL
146.2 Finished test run at Sun Nov 26 06:11:56 2023.
146.2 make: *** [makefile:864: test_harness] Error 1
```

Looking deeper, it seems that the `META.json` isn't regenerated properly:

```
% LD_LIBRARY_PATH=/usr/src/perl ./perl -I. -MTestInit t/porting/regen.t
[...]
# ./perl -Ilib Porting/makemeta -j --tap
not ok - Porting/makemeta META.json
'META.json-new' and 'META.json' differ
[...]

ERROR. There are generated files which are NOT up to date.
You should run the following commands to update these files:

/usr/src/perl/perl -Ilib Porting/makemeta -j

% cp META.json{,.cur}
% LD_LIBRARY_PATH=/usr/src/perl ./perl -Ilib Porting/makemeta -j 
Changed: META.json
root@4210cf83e0fe:/usr/src/perl# diff -Naur META.json{.cur,}
--- META.json.cur       2023-11-26 06:59:36.318324009 +0000
+++ META.json   2023-11-26 07:00:05.470153008 +0000
@@ -106,7 +106,7 @@
          "vxs.inc"
       ]
    },
-   "release_status" : "testing",
+   "release_status" : "stable",
    "resources" : {
       "bugtracker" : {
          "web" : "https://github.com/Perl/perl5/issues"

```

Looks like it was forgotten to update https://github.com/Perl/perl5/blob/8fddb434ae0d0353473456561ec6b844580ff726/META.json#L109

**Comment by zakame** [comment]
I see there is `5.38.1a` /`5.36.2a`/`5.34.2a` releases fixing this (e.g. https://metacpan.org/release/PEVANS/perl-5.38.1a/diff/PEVANS%2Fperl-5.38.1) - should there be a follow-up announcement?

**Comment by leonerd** [comment]
Those are an attempt to fix it, however some of those themselves have errors.

At that point in the afternoon yesterday I realised I'd messed up twice and wasn't in a good frame of mind to look into fixing it, so I stepped away from keyboard to calm down. I'm going to have another look today and see if we can sort it all out.

**Comment by zakame** [comment]
Thanks Paul, no worries! 👍

**Comment by jkeenan** [comment]
*Advance Disclaimer: I have never done a production, RC, dev or maintenance release of perl.  So my analysis of what went wrong may be faulty!*

```
$ git clean -dfx

$ git checkout maint-5.38
Switched to branch 'maint-5.38'
Your branch is up to date with 'origin/maint-5.38'.

$ git show --format=short
commit fc0caa60d8e9cee505d2844995ac04548787fd14 (HEAD -> maint-5.38, origin/maint-5.38)
Author: Paul Evans <leonerd@leonerd.org.uk>

    Update META.json using the freshly-built perl so the JSON::PP version number it writes inside is correct

commit 651571e1ea6c054668459fd5fabc299a75a1fc6e
Author: Paul Evans <leonerd@leonerd.org.uk>

    Update META.json for real release

commit 8fddb434ae0d0353473456561ec6b844580ff726 (tag: v5.38.1)
Author: Paul Evans <leonerd@leonerd.org.uk>

    Disarm RC1

commit bbb63412cd349e65265cd0ad7935191bc7013a41
Author: Paul Evans <leonerd@leonerd.org.uk>

    Bump patchlevel to RC1

...
```
Everything looked good at this point, where @leonerd was releasing perl-5.38.1-RC1.  So I created a branch at that commit.
```
$ git checkout bbb63412cd349e65265cd0ad7935191bc7013a41
$ git switch -c gh-21761-exploration-20231126-jk

$ git show --format=fuller |cat
commit bbb63412cd349e65265cd0ad7935191bc7013a41
Author:     Paul Evans <leonerd@leonerd.org.uk>
AuthorDate: Tue Nov 21 16:31:08 2023 +0000
Commit:     Paul Evans <leonerd@leonerd.org.uk>
CommitDate: Tue Nov 21 16:39:32 2023 +0000

    Bump patchlevel to RC1

diff --git a/META.json b/META.json
index 9adf86c763..44c72ee956 100644
--- a/META.json
+++ b/META.json
@@ -106,7 +106,7 @@
          "vxs.inc"
       ]
    },
-   "release_status" : "stable",
+   "release_status" : "testing",
    "resources" : {
       "bugtracker" : {
          "web" : "https://github.com/Perl/perl5/issues"
diff --git a/patchlevel.h b/patchlevel.h
index 0972100b96..cd2a4f187e 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -158,6 +158,7 @@ hunk.
 #  endif
 static const char * const local_patches[] = {
         NULL
+	,"RC1"
 #ifdef PERL_GIT_UNCOMMITTED_CHANGES
         ,"uncommitted-changes"
 #endif

```
So now we have a branch which replicates where @leonerd was when he released RC1 on Nov 21.  To convert this from an RC release to a production release, I believe the next step -- per lines 885-902 of `Porting/release_managers_guide.pod` -- should have been to **re-edit** `patchlevel.h` to remove the line containing `RC`.
```
$ vi +161 patchlevel.h
```
Delete line 161; save file
```
$ git diff
diff --git a/patchlevel.h b/patchlevel.h
index cd2a4f187e..0972100b96 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -158,7 +158,6 @@ hunk.
 #  endif
 static const char * const local_patches[] = {
         NULL
-       ,"RC1"
 #ifdef PERL_GIT_UNCOMMITTED_CHANGES
         ,"uncommitted-changes"
 #endif

$ git add patchlevel.h

$ git commit -m "Remove RC1 from patchlevel.h"
[gh-21761-exploration-20231126-jk 4ee452734f] Remove RC1 from patchlevel.h
 1 file changed, 1 deletion(-)
 
$ git show --format=short
commit 4ee452734f9c74393a7561fc85d8b17b909e9663 (HEAD -> gh-21761-exploration-20231126-jk)
Author: James E Keenan <jkeenan@cpan.org>

    Remove RC1 from patchlevel.h

diff --git a/patchlevel.h b/patchlevel.h
index cd2a4f187e..0972100b96 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -158,7 +158,6 @@ hunk.
 #  endif
 static const char * const local_patches[] = {
         NULL
-       ,"RC1"
 #ifdef PERL_GIT_UNCOMMITTED_CHANGES
         ,"uncommitted-changes"
 #endif
```
Now, I have to build a perl at this commit, before I can run `./perl -Ilib Porting/makemeta` to update the metadata.
```
$ sh ./Configure -des && make test_prep
...

$ ./perl -Ilib Porting/makemeta
Changed: META.json

$ git diff
diff --git a/META.json b/META.json
index 44c72ee956..9adf86c763 100644
--- a/META.json
+++ b/META.json
@@ -106,7 +106,7 @@
          "vxs.inc"
       ]
    },
-   "release_status" : "testing",
+   "release_status" : "stable",
    "resources" : {
       "bugtracker" : {
          "web" : "https://github.com/Perl/perl5/issues"

$ git add META.json
$ git commit -m "Restore 'release_status' to 'stable'
$ TEST_JOBS=1 make test_porting
...
All tests successful.
...
Result: PASS
```
Subsequent testing per lines 912-937 of `Porting/release_managers_guide.pod` produced good results.  We would then proceed to generating an annotated release tag and building a tarball.

As I get time, I will see if this analysis applies to perl-5.36.2 and perl-5.34.2.

**Comment by zakame** [comment]
Now fixed in [5.38.2/5.36.3/5.34.3](https://www.nntp.perl.org/group/perl.perl5.porters/2023/11/msg267400.html), thanks again @leonerd ! 🎉
